### PR TITLE
Setup circleci for 1.7

### PR DESCRIPTION
**Pull request by anaelChardan**
**Description (for Contributor and Core Developer)**

22 tests are failing. Merging.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      - run:
+          name: Change owner on project dir (default user = circleci (1001) and docker needs uid 1000)
+          command: sudo chown -R 1000:1000 ../project
+      - run:
+          name: Start containers
+          command: docker-compose up -d
+      - restore_cache:
+          name: Restore cache - vendor
+          keys:
+            - vendor-v1-{{ checksum "composer.lock" }}
```
Comment (damien-carcel) [inline .circleci/config.yml]: Add another entry with `- vendor-v1-`.
It provides a fallback for the cache when the content of composer.lock changes.

What's more, there is no composer.lock in the PIM, so not really usable, no?

File: behat.yml.dist [hunk 2]
```diff
@@ -4,19 +4,21 @@ default:
     context:
         class:  Context\FeatureContext
         parameters:
-            base_url: http://akeneo-pim-behat.local/
+            base_url: http://akeneo-behat/
```
Comment (damien-carcel) [inline behat.yml.dist]: I would put back `127.0.0.1` and use your `pim-setup.sh` (like I think it is done in 2.x)

File: composer.json [hunk 3]
```diff
@@ -80,7 +80,7 @@
         "sebastian/exporter": "2.0.*",
         "akeneo/phpspec-skip-example-extension": "2.0.*",
         "akeneo/php-coupling-detector": "0.2",
-        "liuggio/fastest": "1.6.*"
+        "liuggio/fastest": "1.5.*"
```
Comment (damien-carcel) [inline composer.json]: why do you need that?

Comment (anaelChardan) [reply composer.json]: To list scenarios, this version isn't compatible with our behat version it looks like we've never used it :D

File: behat.yml.dist (hunk 2 above)
Comment (anaelChardan) [reply behat.yml.dist]: I dunno, this version will die soon and as everyone have to change it for now :/

File: .circleci/config.yml (hunk 1 above)
Comment (anaelChardan) [reply .circleci/config.yml]: I dunno, could you show an example? ;)

File: behat.yml.dist (hunk 2 above)
Comment (damien-carcel) [reply behat.yml.dist]: Acceptable argument indeed.

File: .circleci/config.yml (hunk 1 above)
Comment (damien-carcel) [reply .circleci/config.yml]: Why do you need an example? This is how the cache system of Circle is documented. The file you make the checksum has to exist, or it doesn't have any sense.

For the fall back, it works as follow: https://github.com/damien-carcel/app-skeleton/blob/master/.circleci/config.yml#L35

File: composer.json (hunk 3 above)
Comment (damien-carcel) [reply composer.json]: ><

File: .circleci/config.yml (hunk 1 above)
Comment (damien-carcel) [reply .circleci/config.yml]: In any case, I think you should commit your composer.lock.
Or you use `composer.json` for the checksum, but it is less usefull, as it is not supposed to change in time, so the checksum used will never change either.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+      - run:
+          name: Pim database installation
+          command: docker-compose exec akeneo-behat app/console --env=behat pim:install --force --clean
+      - run:
+            name: Change owner on project dir after installing when there is no cache
+            command: sudo chmod -R 777 ../project
+      - save_cache:
+          paths:
+            - ./vendor
+          key: vendor-v7-{{ checksum "composer.lock" }}
```
Comment (damien-carcel) [inline .circleci/config.yml]: You store a `v7` here, but restore a `v1` just above. Not gonna work ;-)

Comment (anaelChardan) [reply .circleci/config.yml]: Copy paste from the 2.3 X)

File: .circleci/config.yml (hunk 1 above)
Comment (anaelChardan) [reply .circleci/config.yml]: I will remove the cache

**Pull request merged**